atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/TransactionScopedCache.java

```diff
@@ -60,11 +61,21 @@ Map<Cell, byte[]> get(
             Set<Cell> cells,
             Function<Set<Cell>, ListenableFuture<Map<Cell, byte[]>>> valueLoader);
 
+    Map<Cell, byte[]> getWithCachedRef(
+            TableReference tableReference,
+            Set<Cell> cells,
+            Function<CacheLookupResult, ListenableFuture<Map<Cell, byte[]>>> valueLoader);
```

Comment:
I could have also just replaced the existing `get` and `getAsync` to receive the `CacheLookupResult` instead of creating new two separate methods. Let me know what you prefer.

Reply:
Not very, tbh. Although this PR is already getting quite big, so I'll try to do the refactor in a separate one.